src/customisations/WidgetPermissions.ts

```diff
@@ -36,11 +36,24 @@ async function preapproveCapabilities(
     return new Set(); // no additional capabilities approved
 }
 
+/**
+ * Approves the widget for identity token request, if it can be approved.
+ * This will be used to give certain widgets identity token without having to
+ * prompt the user to approve it.
+ * @param {Widget} widget The widget to approve identity request for.
+ * @returns {boolean} Resolves to true if identity request is preapproved, false otherwise
+ */
+async function isIdentityRequestPreapproved(widget: Widget): Promise<boolean> {
+    return false;
+}
```

Comment:
as I understand it, this function is never called, and only exists to provide the type of `IWidgetPermissionCustomisations.isIdentityRequestPreapproved` ? It seems a very round-the-houses way of defining a type.

Reply:
You are correct, this function is not called and just an example for how the customization can be implemented.  During build time it is replaced with another function if desired, so `isIdentityRequestPreapproved` would really never be used.

As all other customizations are implement this way, e.g. the `preapproveCapabilities` function above, so I think we should keep the pattern?

Reply:
are there any other examples of this than `preapproveCapabilities`?

At the very least, please add a comment to the function saying it is never called and is only an example.

Reply:
Function is updated with the comment that it is used to provide the type only.

`src/customisations` contains more similar use cases, for example: 'src/customisations/Security.ts'